sourcecode/hub/app/Http/Controllers/GoogleController.php

```diff
@@ -0,0 +1,41 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\Models\User;
+use Illuminate\Support\Facades\Auth;
+use Illuminate\Support\Facades\Hash;
+use Laravel\Socialite\Facades\Socialite;
+use Symfony\Component\HttpFoundation\RedirectResponse;
+
+class GoogleController extends Controller
+{
+    public function loginWithGoogle(): RedirectResponse
+    {
+        return Socialite::driver('google')->redirect();
+    }
+
+    public function callbackFromGoogle(): RedirectResponse
+    {
+        try {
```

Comment:
This try/catch is completely useless, as it does the exact same as if there hadn't been a try/catch here.